src/main/java/uk/gov/pay/api/model/CreateDirectDebitPaymentRequest.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.pay.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import uk.gov.pay.api.utils.JsonStringBuilder;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Size;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+@ApiModel(value = "CreatePaymentRequest", description = "The Payment Request Payload")
+public class CreateDirectDebitPaymentRequest extends CreatePaymentRequest{
+
+    public static final int AGREEMENT_ID_MAX_LENGTH = 26;
+    public static final String AGREEMENT_ID_FIELD_NAME = "agreement_id";
+
+    @Size(max = AGREEMENT_ID_MAX_LENGTH, message = "Must be less than or equal to {max} characters length")
+    @JsonProperty(value = AGREEMENT_ID_FIELD_NAME)
+    private final String agreementId;
+
+
+    public CreateDirectDebitPaymentRequest(CreatePaymentRequestBuilder createPaymentRequestBuilder) {
+        super(createPaymentRequestBuilder);
+        this.agreementId = createPaymentRequestBuilder.getAgreementId();
+    }
+
+    @Override
+    @Min(value = 1, message = "Must be greater than or equal to 1")
+    public int getAmount() {
+        return super.getAmount();
+    }
+
+    @ApiModelProperty(value = "ID of the agreement being used to collect the payment", required = false, example = "33890b55-b9ea-4e2f-90fd-77ae0e9009e2")
+    public Optional<String> getAgreementId() {
```

Comment:
An create on-demand Direct Debit payment request should always have an agreement ID

Reply:
yep 👍